internal/controller/deployment_ to_pdb_controller.go

```diff
@@ -0,0 +1,166 @@
+		}
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+	log.Info("Found: ", "deployment", deployment.Name, "namespace", deployment.Namespace)
+	// If the Deployment is created, ensure a PDB exists
+	return r.handleDeploymentCreation(ctx, &deployment)
+}
+
+// handleDeploymentCreation creates a PodDisruptionBudget when a Deployment is created or updated.
+func (r *DeploymentToPDBReconciler) handleDeploymentCreation(ctx context.Context, deployment *v1.Deployment) (reconcile.Result, error) {
+	log := log.FromContext(ctx)
+	// Check if PDB already exists for this Deployment
+	pdb := &policyv1.PodDisruptionBudget{}
+	err := r.Get(ctx, client.ObjectKey{
+		Namespace: deployment.Namespace,
+		Name:      deployment.Name + "-pdb",
+	}, pdb)
+
+	if err == nil {
+		// PDB already exists, nothing to do
+		log.Info("PodDisruptionBudget already exists", "namespace", deployment.Namespace, "name", deployment.Name)
+		//if pdb.Spec.MinAvailable.IntVal != *deployment.Spec.Replicas {
+		//	pdb.Spec.MinAvailable.IntVal = *deployment.Spec.Replicas
+		//	if err := r.Update(ctx, pdb); err != nil {
+		//		return reconcile.Result{}, err
+		//	}
+		//}
+		return reconcile.Result{}, nil
+	}
+
+	// Create a new PDB for the Deployment
+	pdb = &policyv1.PodDisruptionBudget{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PodDisruptionBudget",
+			APIVersion: "policy/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.generatePDBName(deployment.Name),
+			Namespace: deployment.Namespace,
+			Annotations: map[string]string{
+				"createdBy": "DeploymentToPDBController",
+				"target":    deployment.Name,
+			},
+		},
+		Spec: policyv1.PodDisruptionBudgetSpec{
+			MinAvailable: &intstr.IntOrString{IntVal: *deployment.Spec.Replicas},
+			Selector:     &metav1.LabelSelector{MatchLabels: deployment.Spec.Selector.MatchLabels},
+		},
+	}
+
+	log.Info("Creating PodDisruptionBudget", "namespace", pdb.Namespace, "name", pdb.Name)
+	if err := r.Create(ctx, pdb); err != nil {
+		log.Error(err, "unable to create PDB")
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *DeploymentToPDBReconciler) generatePDBName(deploymentName string) string {
+	return deploymentName + "-pdb"
+}
+
+// handleDeploymentDeletion deletes the associated PDB when the Deployment is deleted
+func (r *DeploymentToPDBReconciler) handleDeploymentDeletion(ctx context.Context, req ctrl.Request) (reconcile.Result, error) {
+	log := log.FromContext(ctx)
+
+	// Try to get the associated PDB
+	pdb := &policyv1.PodDisruptionBudget{}
+	err := r.Get(ctx, client.ObjectKey{
+		Namespace: req.NamespacedName.Namespace,
+		Name:      r.generatePDBName(req.NamespacedName.Name),
+	}, pdb)
+
+	if err != nil {
+		// If there's no PDB or it can't be fetched, just return
+		log.Info("PodDisruptionBudget does not exist or error fetching", "namespace", req.NamespacedName.Namespace, "name", req.NamespacedName.Name)
+		return reconcile.Result{}, client.IgnoreNotFound(err)
```

Comment:
didn't know client.IgnoreNotFound( was! I  think shoul;d remove some lines in the other reconcilers

Reply:
might be no  have a pdb  if there was one from the user we don't want to conflict with.